src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/LimitedUseItem.java

```diff
@@ -0,0 +1,156 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.apache.commons.lang.Validate;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.bukkit.persistence.PersistentDataContainer;
+import org.bukkit.persistence.PersistentDataType;
+
+import io.github.thebusybiscuit.cscorelib2.chat.ChatColors;
+import io.github.thebusybiscuit.slimefun4.api.SlimefunAddon;
+import io.github.thebusybiscuit.slimefun4.api.items.ItemState;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.items.magical.staves.StormStaff;
+import io.github.thebusybiscuit.slimefun4.utils.LoreBuilder;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+
+/**
+ * This class represents an item with a limited number of uses.
+ * When the item runs out of "uses", it breaks.
+ *
+ * @author Linox
+ * @author Walshy
+ * @author TheBusyBiscuit
+ * @author martinbrom
+ *
+ * @see StormStaff
+ */
+public abstract class LimitedUseItem extends SimpleSlimefunItem<ItemUseHandler> {
+
+    private static final NamespacedKey usageKey = new NamespacedKey(SlimefunPlugin.instance(), "uses_left");
+    private static final Pattern REGEX = Pattern.compile(ChatColors.color("&e[0-9]+ Uses &7left"));
```

Comment:
These do **not** need to be final.

```suggestion
    private final NamespacedKey usageKey = new NamespacedKey(SlimefunPlugin.instance(), "uses_left");
    private final Pattern lorePattern = Pattern.compile(ChatColors.color("&e[0-9]+ Uses &7left"));
```

Reply:
The comment says `final` but the suggestion removes the `static` keyword. Assuming the suggestion is correct, why tho? There will often be multiple instances of the `LimitedUseItem` and the values do not differ. It is safe for them to be static then, is it not?

Reply:
1. `static` fields create a potential risk of memory leakage due to the way Bukkit's ClassLoader works, so it is best to keep those to a minimum.
2. Initialising the `NamespacedKey` static-ly is an incedribly bad idea. We do not have explicit control over when the class is loaded into memory and therefore no control over when this field is initialised. `SlimefunPlugin.instance()` may be `null` at that point, making this a class member ensures that it only gets loaded when we explicitly instantiate an object of that class.

The `Pattern` might make sense to instantiate static-ly but I would suggest this to be moved to `PatternUtils` then, so we have finer control over where our static fields lay.
Also `usageKey` may perhaps better be named `defaultUsageKey`, as it can be overridden by sub classes.